dispatcher_plugin_nb2workflow/products.py

```diff
@@ -72,25 +73,25 @@ def _init_as_list(cls, encoded_data, *args, **kwargs):
     @classmethod
     def prod_list_factory(cls, output_description_dict, output, out_dir = None, ontology_path = None):
         par_prod_classes = parameter_products_factory(ontology_path)
-        
-        mapping = {x.type_key: x for x in cls.__subclasses__() + par_prod_classes} 
+
+        mapping = {x.type_key: x for x in cls.__subclasses__() + par_prod_classes if hasattr(x, 'type_key')}
         
         prod_list = []
         for key in output_description_dict.keys():
             owl_type = output_description_dict[key]['owl_type']
-            
+
             extra_kw = {}
             extra_ttl = output_description_dict[key].get('extra_ttl')
-            if extra_ttl == '\n': extra_ttl = None 
+            if extra_ttl == '\n': extra_ttl = None
             if extra_ttl:
                 extra_kw = {'extra_ttl': extra_ttl}
 
             try:
-                prod_list.extend( mapping.get(owl_type, cls)._init_as_list(output[key], 
-                                                                           out_dir=out_dir, 
-                                                                           name=key, 
+                prod_list.extend( mapping.get(owl_type, cls)._init_as_list(output[key],
+                                                                           out_dir=out_dir,
+                                                                           name=key,
                                                                            **extra_kw
-                                                                           ) 
+                                                                           )
```

Comment:
a bunch of changes here just add spaces

Reply:
probably introduced by the IDEs during some copy/paste operations

Reply:
this code was already there, strange that you needed to copy-paste it. But anyway, could remove the copy-paste artifacts.